config.m4

```diff
@@ -206,7 +206,6 @@ if test "$MONGODB" != "no"; then
     bson-atomic.c \
     bson-clock.c \
     bson-context.c \
-    bson-decimal128.c \
```

Comment:
He was mistaken. Both this and the client metadata sources in libmongoc are missing an `#ifdef` wrap.